specfile/specfile.py

```diff
@@ -0,0 +1,101 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import contextlib
+import types
+from pathlib import Path
+from typing import Iterator, List, Optional, Tuple, Type, Union
+
+from specfile.rpm import RPM, Macros
+from specfile.sections import Sections
+from specfile.tags import Tags
+
+
+class Specfile:
+    """
+    Class representing spec file.
+
+    Attributes:
+        path: Path to the spec file.
+        sourcedir: Path to sources and patches.
+        autosave: Whether to automatically save any changes made.
+        macros: List of macro definitions.
+    """
+
+    def __init__(
+        self,
+        path: Union[Path, str],
+        sourcedir: Optional[Union[Path, str]] = None,
+        autosave: bool = False,
+        macros: Optional[List[Tuple[str, str]]] = None,
+    ) -> None:
+        self.path = Path(path)
+        self.sourcedir = Path(sourcedir or self.path.parent)
+        self.autosave = autosave
+        self.macros = macros.copy() if macros is not None else None
+        self._sections = Sections.parse(self.path.read_text())
+        self._spec = RPM.parse(str(self._sections), self.sourcedir, self.macros)
+        self._parsed_sections = Sections.parse(self._spec.parsed)
+
+    def __enter__(self) -> "Specfile":
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[types.TracebackType],
+    ) -> None:
+        if not self.autosave:
+            self.save()
```

Comment:
Is the condition OK? (should the `not` be there?)

Reply:
The point of the condition is not to unnecessarily write to the spec file multiple times (at this point it should have been already saved if a change was made), but perhaps it's better to save no matter what.